datashape/coretypes.py

```diff
@@ -526,18 +455,10 @@ def __init__(self, *parameters, **kwds):
             raise ValueError('the data shape should be constructed from 2 or'
                              ' more parameters, only got %s' % len(parameters))
         self.composite = True
+        self.name = kwds.get('name')
 
-        name = kwds.get('name')
-        if name:
-            self.name = name
-            self.__metaclass__._registry[name] = self
-        else:
-            self.name = None
-
-        ###
-        # TODO: Why are low-level concepts like strides and alignment on
-        # TODO: the datashape?
-        ###
+        if self.name is not None:
```

Comment:
i thought about removing this feature entirely, but it's used quite heavily in `CType`